pkg/chains/signing.go

```diff
@@ -55,6 +56,29 @@ type ObjectSigner struct {
 	Recorder MetricsRecorder
 }
 
+type ErrorTypeMetricValue string
+
+const (
+	// PayloadCreationError is recorded when CreatePayload fails.
+	PayloadCreationError ErrorTypeMetricValue = "payload_creation"
+	MarshalPayloadError  ErrorTypeMetricValue = "marshal_payload"
+	SigningError         ErrorTypeMetricValue = "signing"
+	StorageError         ErrorTypeMetricValue = "storage"
+	TlogError            ErrorTypeMetricValue = "tlog"
+)
+
+// shouldRecordError returns true if the artifact type is TaskRunArtifact or PipelineRunArtifact.
+func (o *ObjectSigner) shouldRecordError(kind string) bool {
```

Comment:
I'm a bit confused with the name of function name and the logic inside the function